Assets/MixedRealityToolkit/Utilities/StandardShader/BaseMeshOutline.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities
+{
+    /// <summary>
+    /// Abstract component to encapsulate common functionality around outline components. 
+    /// </summary>
+    [HelpURL("https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/README_MRTKStandardShader.html#mesh-outlines")]
+    public abstract class BaseMeshOutline : MonoBehaviour
+    {
+        /// <summary>
+        /// The material used to render the outline. Outline materials should normal have "Depth Write" set to Off and "Vertex Extrusion" enabled.
+        /// Most MRTK/Standard features should work as an outline material, but it is recommended to use a simple unlit material.
+        /// </summary>
+        public Material OutlineMaterial
+        {
+            get { return outlineMaterial; }
+            set
+            {
+                if (outlineMaterial != value)
+                {
+                    outlineMaterial = value;
+                    ApplyOutlineMaterial();
+                }
+            }
+        }
+
+        [Tooltip("The material used to render the outline. Outline materials should normal have \"Depth Write\" set to Off and \"Vertex Extrusion\" enabled.")]
+        [SerializeField]
+        protected Material outlineMaterial = null;
+
+        /// <summary>
+        /// How thick (in meters) should the outline be. Overrides the "Extrusion Value" in the MRTK/Standard material.
+        /// </summary>
+        public float OutlineWidth
+        {
+            get { return outlineWidth; }
+            set
+            {
+                if (outlineWidth != value)
+                {
+                    outlineWidth = value;
+                    ApplyOutlineWidth();
+                }
+            }
+        }
+
+        [Tooltip("How thick (in meters) should the outline be. Overrides the \"Extrusion Value\" in the MRTK/Standard material.")]
+        [SerializeField]
+        [Range(0.001f, 1.0f)]
+        protected float outlineWidth = 0.01f;
+
+        #region MonoBehaviour Implementation
+
+        /// <summary>
+        /// Enables users to modify inspector properties while playing in the editor.
+        /// </summary>
+        protected virtual void OnValidate()
+        {
+            ApplyOutlineMaterial();
```

Comment:
Why use onvalidate instead of a BaseMesh inspector class? This is for the component right?

Reply:
I'm just using OnValidate to apply changes as they occur in the editor. I could make a customer inspector but seems a little overkill?